dht/token.cc

```diff
@@ -286,4 +286,35 @@ dht::token token::from_int64(int64_t i) {
     return {kind::key, i};
 }
 
+static
+dht::token find_first_token_for_shard_in_not_wrap_around_range(const dht::sharder& sharder, dht::token start, dht::token end, size_t shard_idx) {
+    // Invariant start < end
+    // It is guaranteed that start is not MAX_INT64 because end is greater
+    auto t = dht::token::from_int64(dht::token::to_int64(start) + 1);
+    if (sharder.shard_of(t) != shard_idx) {
+        t = sharder.token_for_next_shard(t, shard_idx);
+    }
+    return std::min(t, end);
```

Comment:
This is returning the wrong result, no? Maybe return std::nullopt instead (and std::optional<token> for the type).

Reply:
Returning optional instead.